cts/cts-cli.in

```diff
@@ -553,6 +581,17 @@ function test_tools() {
     cmd="crm_resource --clear --expired"
     test_assert $CRM_EX_OK
 
+    # Clear has already been tested elsewhere, but we need to get rid of the
+    # constraints so testing delete works.  It won't delete if there's still
+    # a reference to the resource somewhere.
+    desc="Clear all implicit constraints for dummy"
+    cmd="crm_resource -r dummy -U"
+    test_assert $CRM_EX_OK
+
+    desc="Delete a resource"
+    cmd="crm_resource -D -r dummy -t primitive"
+    test_assert $CRM_EX_OK
+
```

Comment:
The cts-* regression tests have to be able to operate when the cluster is not running. CIB_file allows us to get around that for the CIB, but everything else that needs something from the cluster just can't be regression tested. Theoretically CTS proper could test those tool options, but it doesn't currently.

The cts-cli and cts-scheduler regression tests additionally have to be able to run from a source checkout. The cts-exec and cts-fencing regression tests start their respective daemons in a "standalone" mode to be able to exercise their IPC APIs.

Reply:
Ah, got it.  The tests I added do work, so I was a little surprised that the others did not.  Like I said in the original comment, these tests did find bugs so I have reason to believe that additional tests would find additional bugs.  I guess I will just have to try them manually.

Reply:
I remembered another possibility: we ran into this when testing --refresh, and added a "dry run" mode for it. See "Dry run" in crm_resource.c. All it does is skip the real work when CIB_file is set. We could make other commands do the same. However that obviously doesn't actually test the actual functionality, it's just to test argument parsing etc. and allow test scripts to work on a file, so probably not worth it for the regression tests.